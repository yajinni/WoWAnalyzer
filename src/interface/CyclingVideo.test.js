import React from 'react';
import { shallow } from 'enzyme';

import CyclingVideo from './CyclingVideo';

const FAKE_VIDEOS = [
  '1',
  '2',
  '3',
];

describe('CyclingVideo', () => {
  it('matches snapshot', () => {
    const tree = shallow(
      <CyclingVideo videos={FAKE_VIDEOS} randomValue={0} />,
    );
    expect(tree).toMatchSnapshot();
  });
  it('passes unrecognized props', () => {
    const tree = shallow(
      <CyclingVideo videos={FAKE_VIDEOS} randomValue={0} style={{ color: 'red' }} />,
    );
    expect(tree).toMatchSnapshot();
  });
  it('uses the random value to determine at which video to start', () => {
    {
      const comp = shallow(
        <CyclingVideo videos={FAKE_VIDEOS} randomValue={0} />,
      );
      expect(comp.find('source').prop('src')).toBe('1');
    }
    {
      const comp = shallow(
        <CyclingVideo videos={FAKE_VIDEOS} randomValue={0.5} />,
      );
      expect(comp.find('source').prop('src')).toBe('2');
    }
    {
      const comp = shallow(
        <CyclingVideo videos={FAKE_VIDEOS} randomValue={0.999} />,
      );
      expect(comp.find('source').prop('src')).toBe('3');
    }
  });
  it('starts the next video when the current ends', () => {
    const comp = shallow(
      <CyclingVideo videos={FAKE_VIDEOS} randomValue={0} />,
    );
    expect(comp.find('source').prop('src')).toBe('1'); // sanity
    comp.simulate('ended');
    expect(comp.find('source').prop('src')).toBe('2');
  });
  it('restarts when at the end of the list', () => {
    const comp = shallow(
      <CyclingVideo videos={FAKE_VIDEOS} randomValue={0.999} />,
    );
    expect(comp.find('source').prop('src')).toBe('3'); // sanity
    comp.simulate('ended');
    expect(comp.find('source').prop('src')).toBe('1');
  });
});
