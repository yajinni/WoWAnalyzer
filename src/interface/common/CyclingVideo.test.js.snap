// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CyclingVideo matches snapshot 1`] = `
<video
  autoPlay={true}
  key="1"
  muted={true}
  onEnded={[Function]}
  randomValue={0}
>
  <source
    src="1"
    type="video/mp4"
  />
</video>
`;

exports[`CyclingVideo passes unrecognized props 1`] = `
<video
  autoPlay={true}
  key="1"
  muted={true}
  onEnded={[Function]}
  randomValue={0}
  style={
    Object {
      "color": "red",
    }
  }
>
  <source
    src="1"
    type="video/mp4"
  />
</video>
`;
